src/modulefinder/sentry_modulefinder_windows.c

```diff
@@ -44,36 +44,48 @@ extract_pdb_info(uintptr_t module_addr, sentry_value_t module)
         nt_headers->OptionalHeader.SizeOfImage);
     sentry_value_set_by_key(module, "code_id", sentry_value_new_string(id_buf));
 
-    uint32_t relative_addr
-        = nt_headers->OptionalHeader.DataDirectory[IMAGE_DIRECTORY_ENTRY_DEBUG]
-              .VirtualAddress;
+    IMAGE_DATA_DIRECTORY debug_entry
+        = nt_headers->OptionalHeader.DataDirectory[IMAGE_DIRECTORY_ENTRY_DEBUG];
+
+    size_t relative_addr = (size_t)debug_entry.VirtualAddress;
     if (!relative_addr) {
         return;
     }
 
-    PIMAGE_DEBUG_DIRECTORY debug_dict
-        = (PIMAGE_DEBUG_DIRECTORY)(module_addr + relative_addr);
-    if (!debug_dict || debug_dict->Type != IMAGE_DEBUG_TYPE_CODEVIEW) {
+    size_t table_size = (size_t)debug_entry.Size;
+    size_t entry_size = sizeof(IMAGE_DEBUG_DIRECTORY);
+    if (table_size % entry_size != 0) {
```

Comment:
Would it be possible to somehow iterate over some entries and just drop the one that was somehow malformed and has fewer bytes than necessary? Or is it not possible to tell which one would even be broken at this point?

Reply:
I think doing the same thing as crashpad here is okay. These structs have a fixed size, and the table should always be a multiple of that, otherwise the compiler did something wrong